execution_engine_testing/tests/src/test/system_contracts/auction/distribute.rs

```diff
@@ -1283,90 +1268,97 @@ fn should_distribute_reinvested_rewards_by_different_factor() {
-            VALIDATOR_2.clone(),
-            VALIDATOR_2_REWARD_FACTOR_1,
-        ))
-        .with_reward_item(RewardItem::new(
-            VALIDATOR_3.clone(),
-            VALIDATOR_3_REWARD_FACTOR_1,
-        ))
-        .with_next_era_id(builder.get_era().successor())
-        .with_run_auction(true)
-        .build();
+    let distribute_request = ExecuteRequestBuilder::contract_call_by_hash(
+        *SYSTEM_ADDR,
+        builder.get_auction_contract_hash(),
+        METHOD_DISTRIBUTE,
+        runtime_args! {
+            ARG_ENTRY_POINT => METHOD_DISTRIBUTE,
+            ARG_VALIDATOR => VALIDATOR_1.clone()
+        },
+    )
+    .build();
 
-    builder
-        .step(step_request)
-        .expect("must execute step successfully");
+    builder.exec(distribute_request).commit().expect_success();
 
-    let validator_1_staked_amount_1 = *get_validator_bid(&mut builder, VALIDATOR_1.clone())
-        .expect("should have validator bid")
-        .staked_amount();
+    let delegators_share = {
+        let commission_rate = Ratio::new(
+            U512::from(VALIDATOR_1_DELEGATION_RATE),
+            U512::from(DELEGATION_RATE_DENOMINATOR),
+        );
+        let reward_multiplier =
+            Ratio::new(U512::from(TOTAL_DELEGATOR_STAKE), U512::from(TOTAL_STAKE));
+        let delegator_reward = expected_total_reward
+            .checked_mul(&reward_multiplier)
+            .expect("must get delegator reward");
+        let commission = delegator_reward
+            .checked_mul(&commission_rate)
+            .expect("must get commission");
+        delegator_reward.checked_sub(&commission).unwrap()
+    };
 
-    let validator_2_staked_amount_1 = *get_validator_bid(&mut builder, VALIDATOR_2.clone())
-        .expect("should have validator bid")
-        .staked_amount();
+    let delegator_1_expected_payout = {
+        let reward_multiplier = Ratio::new(
+            U512::from(DELEGATOR_1_STAKE),
+            U512::from(TOTAL_DELEGATOR_STAKE),
+        );
+        delegators_share
+            .checked_mul(&reward_multiplier)
+            .map(|ratio| ratio.to_integer())
+            .unwrap()
+    };
 
-    let validator_3_staked_amount_1 = *get_validator_bid(&mut builder, VALIDATOR_3.clone())
-        .expect("should have validator bid")
-        .staked_amount();
+    let delegator_2_expected_payout = {
+        let reward_multiplier = Ratio::new(
+            U512::from(DELEGATOR_2_STAKE),
+            U512::from(TOTAL_DELEGATOR_STAKE),
+        );
+        delegators_share
+            .checked_mul(&reward_multiplier)
+            .map(|ratio| ratio.to_integer())
+            .unwrap()
+    };
 
-    let validator_1_actual_payout_1 = {
+    let validator_1_expected_payout = {
+        let total_delegator_payout = delegator_1_expected_payout + delegator_2_expected_payout;
+        let validators_part = expected_total_reward - Ratio::from(total_delegator_payout);
+        validators_part.to_integer()
+    };
```

Comment:
My personal preference would be to have the expected payouts calculated by hand and hardcoded here - this code seems to be pretty much a copy of the code that is in `distribute_rewards`, so as it is now, we're basically checking the correctness of the code against itself.

Reply:
Nearly all of these tests should be simplified and re-written, but I only fixed them up enough to have them pass.